improver/utilities/spatial.py

```diff
@@ -761,3 +761,55 @@ def add_vicinity_coordinate(
         point, units=units, long_name="radius_of_vicinity", attributes=attributes
     )
     cube.add_aux_coord(coord)
+
+
+def maximum_in_height(
+    cube: Cube, lower_height_bound: float = None, upper_height_bound: float = None
+) -> Cube:
+    """Calculate the maximum value over the height coordinate. If bounds are specified
+    then the maximum value between the lower_height_bound and upper_height_bound is calculated.
+
+    If either the upper or lower bound is None then no bound is applied. For example if no
+    lower bound is provided but an upper bound of 300m is provided then the maximum is
+    calculated for all height levels less than 300m.
+
```

Comment:
Add in that this inherits Iris' meta-data updates to the height coord and to the cell methods.

Reply:
I've added an extra sentence to clarify